runtime/memory.c

```diff
@@ -216,6 +217,14 @@ static int caml_page_table_modify(uintnat page, int toclear, int toset)
 
 #endif
 
+void caml_handle_out_of_memory(void) {
+  if (caml_init_out_of_memory_mode == 0) {
+    caml_raise_out_of_memory();
+  } else {
+    abort();
```

Comment:
You should print an error message before aborting, like we do everywhere else when we use `abort`.